### PR TITLE
Bump FlytePlugins version

### PR DESCRIPTION
**Pull request by pingsutw**
Signed-off-by: Kevin Su <pingsutw@apache.org>

# TL;DR
- Bump flyteplugins version to support ray tasks
- Updated args in `manager.Options{}` since ClientBuilder is removed from the latest version of controller-runtime
- Replace `ConfigMapsResourceLock` with `ConfigMapsLeasesResourceLock` since `ConfigMapsResourceLock` is [removed](https://github.com/kubernetes/client-go/blob/a890e7bc14d5062a2a7eb96a5286239383d5cac8/tools/leaderelection/resourcelock/interface.go#L195)

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [x] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2641

## Follow-up issue
_NA_

### CODE REVIEW

File: pkg/controller/executors/kube.go [hunk 1]
```diff
@@ -96,7 +104,7 @@ type writeThroughCachingWriter struct {
 }
 
 func IDFromObject(obj client.Object, op string) []byte {
-	return []byte(fmt.Sprintf("%s:%s:%s:%s:%s", obj.GetClusterName(), obj.GetObjectKind().GroupVersionKind().String(), obj.GetNamespace(), obj.GetName(), op))
+	return []byte(fmt.Sprintf("%s:%s:%s:%s:%s", obj.GetGenerateName(), obj.GetObjectKind().GroupVersionKind().String(), obj.GetNamespace(), obj.GetName(), op))
```
Comment (hamersaw) [inline pkg/controller/executors/kube.go:107]: Is this going to break backwards compatibility if we deploy while a workflow is running?

File: cmd/controller/cmd/root.go [hunk 2]
```diff
@@ -123,9 +126,11 @@ func executeRootCmd(baseCtx context.Context, cfg *config2.Config) error {
 		limitNamespace = cfg.LimitNamespace
 	}
 	options := manager.Options{
-		Namespace:     limitNamespace,
-		SyncPeriod:    &cfg.DownstreamEval.Duration,
-		ClientBuilder: executors.NewFallbackClientBuilder(propellerScope.NewSubScope("kube")),
+		Namespace:  limitNamespace,
+		SyncPeriod: &cfg.DownstreamEval.Duration,
+		NewClient: func(cache cache.Cache, config *rest.Config, options client.Options, uncachedObjects ...client.Object) (client.Client, error) {
```
Comment (hamersaw) [inline cmd/controller/cmd/root.go:131]: These are required just because of k8s version updates?

File: pkg/controller/executors/kube.go (hunk 1 above)
Comment (pingsutw) [reply pkg/controller/executors/kube.go]: updated it, it should be [GetZZZ_DeprecatedClusterName](https://github.com/kubernetes/apimachinery/blob/97e5df2d0258ac077093867fabf508c1fc31f53b/pkg/apis/meta/v1/meta.go#L175-L177). `GetClusterName()` is remove from latest version of k8s API.

File: cmd/controller/cmd/root.go (hunk 2 above)
Comment (pingsutw) [reply cmd/controller/cmd/root.go:131]: yes, correct.

File: cmd/controller/cmd/root.go [hunk 3]
```diff
@@ -7,6 +7,10 @@ import (
 	"os"
 	"runtime"
 
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
```
Comment (hamersaw) [inline cmd/controller/cmd/root.go:12]: Can we move these so that they're grouped alphabetically like all of them below.

File: cmd/controller/cmd/webhook.go [hunk 4]
```diff
@@ -3,6 +3,10 @@ package cmd
 import (
 	"context"
 
+	"k8s.io/client-go/rest"
```
Comment (hamersaw) [inline cmd/controller/cmd/webhook.go:6]: same imports grouping.

File: pkg/controller/nodes/task/k8s/plugin_manager.go [hunk 5]
```diff
@@ -35,6 +33,8 @@ import (
 	k8stypes "k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/validation"
 
+	"github.com/flyteorg/flyteplugins/go/tasks/errors"
+	nodeTaskConfig "github.com/flyteorg/flytepropeller/pkg/controller/nodes/task/config"
```
Comment (hamersaw) [inline pkg/controller/nodes/task/k8s/plugin_manager.go:37]: is there a reason we moved these from the alphabetized grouping?

File: pkg/controller/executors/kube.go (hunk 1 above)
Comment (hamersaw) [reply pkg/controller/executors/kube.go]: On second thought, this is an in-memory cache. There is no need for backwards compatibility, only uniqueness guarantees. Since this is always empty, let's just drop it completely.

Comment (pingsutw) [reply pkg/controller/executors/kube.go]: That makes sense, thank you. removed it

File: pkg/controller/nodes/task/k8s/plugin_manager.go (hunk 5 above)
Comment (pingsutw) [reply pkg/controller/nodes/task/k8s/plugin_manager.go]: sorry, my IDE auto formatted it. reverted it.

File: cmd/controller/cmd/webhook.go (hunk 4 above)
Comment (pingsutw) [reply cmd/controller/cmd/webhook.go]: fixed

File: cmd/controller/cmd/root.go (hunk 3 above)
Comment (pingsutw) [reply cmd/controller/cmd/root.go]: fixed

File: pkg/controller/executors/kube_test.go [hunk 6]
```diff
@@ -28,18 +28,18 @@ func TestIdFromObject(t *testing.T) {
...
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			p := &v1.Pod{
 				ObjectMeta: metav1.ObjectMeta{
-					Namespace:   tt.args.ns,
-					Name:        tt.args.name,
-					ClusterName: tt.args.cluster,
+					Namespace:                 tt.args.ns,
+					Name:                      tt.args.name,
+					ZZZ_DeprecatedClusterName: tt.args.cluster,
```
Comment (hamersaw) [inline pkg/controller/executors/kube_test.go:42]: We don't need these now right? Or are they required for some reason?

**Review by hamersaw — APPROVED** [review]

**Comment by wild-endeavor** [comment]
@hamersaw do you know why this test runs for so long? is it supposed to take more than six hours?

**Comment by hamersaw** [comment]
> @hamersaw do you know why this test runs for so long? is it supposed to take more than six hours?

@wild-endeavor looks like the codeQL check - looking into it.

**Review by hamersaw — APPROVED** [review]

File: pkg/leaderelection/leader_election.go [hunk 7]
```diff
@@ -41,7 +41,7 @@ func NewResourceLock(corev1 v1.CoreV1Interface, coordinationV1 v12.CoordinationV
 	}
 
 	// Leader id, needs to be unique
-	return resourcelock.New(resourcelock.ConfigMapsResourceLock,
+	return resourcelock.New(resourcelock.ConfigMapsLeasesResourceLock,
```
Comment (honnix) [inline pkg/leaderelection/leader_election.go:44]: I'm not entirely sure, but this change seems to require additional permissions:

```yaml
 apiGroups:
  - coordination.k8s.io
  resources:
  - leases
  verbs: ...
```
If so, this is a breaking change unfortunately.

Comment (honnix) [reply pkg/leaderelection/leader_election.go:44]: In fact, this caused a failed deployment for us because we did not have this RBAC permission in place when upgrading to 1.2.0.

Comment (hamersaw) [reply pkg/leaderelection/leader_election.go:44]: @honnix so sorry about not exposing this in the release notes, thanks for tracking it down - it will be added. Are these additional RBAC permissions an issue?

It looks like the `ConfigMapLeasesResourceLock` has been [the default ](https://github.com/kubernetes-sigs/controller-runtime/blob/1730628f118be1128787012c164379e6ac46c75f/pkg/manager/manager.go#L143-L156 ) for a very long time and we should have migrated over sooner. Now support for `ConfigMapResourceLock` has [been removed](https://github.com/kubernetes/client-go/commit/276ea3ed979947d7cdd4b3d708862245ddcd8883).

Comment (honnix) [reply pkg/leaderelection/leader_election.go:44]: @hamersaw No worries. Just wanted to give you folks a heads-up. It was fairly easy to solve on our side and it is all good now.
